### PR TITLE
requester: use own classes for (re)send dialogues

### PR DESCRIPTION
**Pull request by thc202**
Add classes for the send and resend dialogues, which supersede the core
ones.
Load the send dialogue and add the history reference info when the core
extension is deprecated.
Remove unnecessary methods and the interface `MessageSender`, let the
implementations handle how the messages are sent, it was also exposing
implementation details (e.g. clean up was only needed for HTTP and it
will soon no longer be needed per newer HTTP client).
Rename some methods to be message agnostic and others to be more clear
what they do.

### CODE REVIEW

**Review by kingthorin — APPROVED** [review]

File: addOns/requester/src/main/java/org/zaproxy/addon/requester/internal/SendHttpMessageEditorDialog.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    }
+
+    @Override
+    public void load(ExtensionHook extensionHook) {
+        super.load(extensionHook);
+
+        ZapMenuItem menuItem =
+                new ZapMenuItem(
+                        "menu.tools.manReq",
+                        View.getSingleton().getMenuShortcutKeyStroke(KeyEvent.VK_M, 0, false));
```
Comment (ricekot) [inline addOns/requester/src/main/java/org/zaproxy/addon/requester/internal/SendHttpMessageEditorDialog.java:46]: Doesn't the requester use `ctrl+w`?

File: addOns/requester/src/main/java/org/zaproxy/zap/extension/requester/ExtensionRequester.java [hunk 2]
```diff
@@ -81,7 +90,23 @@ public static ImageIcon getRequesterIcon() {
...
         extensionHook.addOptionsParamSet(getOptionsParam());
+
+        addHrefType(
+                extensionHook,
+                new HrefTypeInfo(
+                        HistoryReference.TYPE_ZAP_USER,
+                        Constant.messages.getString("view.href.type.name.manual"),
+                        hasView() ? getManualIcon() : null));
+
         if (getView() != null) {
```
Comment (ricekot) [inline addOns/requester/src/main/java/org/zaproxy/zap/extension/requester/ExtensionRequester.java:101]: Could use `hasView()`?

File: addOns/requester/src/main/java/org/zaproxy/zap/extension/requester/ManualRequestEditorPanel.java [hunk 3]
```diff
@@ -252,13 +242,16 @@ protected void send(final Message aMessage) {
...
-                        // redraw, as message may have changed after sending
-                        getRequestPanel().updateContent());
+        EventQueue.invokeLater(getMessagePanel()::updateContent);
     }
 
-    protected abstract void saveConfig();
+    /** Saves the configuration of the panel. */
+    public abstract void saveConfig();
 
-    protected abstract HttpPanelRequest getRequestPanel();
```
Comment (ricekot) [inline addOns/requester/src/main/java/org/zaproxy/zap/extension/requester/ManualRequestEditorPanel.java:263]: Won't all inheriting classes also have a response panel?

File: addOns/requester/src/main/java/org/zaproxy/addon/requester/internal/SendHttpMessageEditorDialog.java (hunk 1 above)
Comment (thc202) [reply addOns/requester/src/main/java/org/zaproxy/addon/requester/internal/SendHttpMessageEditorDialog.java:46]: This replaces the core Manual Request Editor, which uses other shortcut.

File: addOns/requester/src/main/java/org/zaproxy/zap/extension/requester/ExtensionRequester.java (hunk 2 above)
Comment (thc202) [reply addOns/requester/src/main/java/org/zaproxy/zap/extension/requester/ExtensionRequester.java:101]: I don't usually update other code unless the patch touches it. IMO we should do these updates in their own PRs and change everything at once.

File: addOns/requester/src/main/java/org/zaproxy/zap/extension/requester/ManualRequestEditorPanel.java (hunk 3 above)
Comment (thc202) [reply addOns/requester/src/main/java/org/zaproxy/zap/extension/requester/ManualRequestEditorPanel.java:263]: No, e.g. WebSocket doesn't have a response.

**Review by psiinon — APPROVED** [review]

**Pull request merged**